cmd/podman/utils/utils.go

```diff
@@ -123,17 +124,14 @@ func IsCheckpointImage(ctx context.Context, namesOrIDs []string) (bool, error) {
 	}
 	imgID := imgData[0].ID
 
-	hostInfo, err := registry.ContainerEngine().Info(ctx)
-	if err != nil {
-		return false, err
-	}
+	runtimeName := filepath.Base(registry.PodmanConfig().RuntimePath)
```

Comment:
RuntimePath is legacy at this point - you want `OCIRuntime`. It should already be the name of the runtime as used by Podman, not a path that needs to be trimmed.

Reply:
Would `OCIRuntime` be set to the value specified with `--runtime` when it is used?

Reply:
I'm not sure what you mean. How would i get the `OCIRuntime`.
For the name, here is what we currently get by default:

hostinfo.OCIRuntime.Name: crun
registry.PodmanConfig().RuntimePath: crun

And with --runtime runc:
hostinfo.OCIRuntime.Name: runc
registry.PodmanConfig().RuntimePath: runc

And with --runtime /usr/bin/crun
hostinfo.OCIRuntime.Name: /usr/bin/crun
registry.PodmanConfig().RuntimePath: /usr/bin/crun

Given the above it seems like the current/old code (using hostinfo) doesn't apply filepath.Base(). Does that mean it is wrong?

Reply:
`PodmanConfig` should be https://github.com/containers/common/blob/main/pkg/config/config.go#L68

In there, you can see that RuntimePath is deprecated (https://github.com/containers/common/blob/main/pkg/config/config.go#L388-L392) while OCIRuntime is the replacement (https://github.com/containers/common/blob/main/pkg/config/config.go#L355-L356)

Reply:
That doesn't seem to give the right thing though.

 If I use ` registry.PodmanConfig().Engine.OCIRuntime`, then by default that contains `crun` which is fine. 

However if i run `podman run --runtime=runc`, it still contains `crun`, which strikes me as wrong.

Reply:
You can't get that from the config... You'd need to grab Libpod's DefaultOCIRuntime to know that.

If RuntimePath in the config is actually reflecting that, that's a bug, not intended behavior.

Reply:
Also, this probably doesn't work properly for podman-remote as implemented, now that I think about it... You'll get the config from the client, not the server.

Reply:
That is what I said above, this needs to be checked on the server side not the client.